src/Simulation/Core/Submitters/SubmissionOptions.cs

```diff
@@ -5,32 +5,37 @@
 
 #nullable enable
 
+namespace System.Runtime.CompilerServices
+{
+    public class IsExternalInit {}
+}
+
 namespace Microsoft.Quantum.Runtime.Submitters
 {
     /// <summary>
     /// Options for a job submitted to Azure Quantum.
     /// </summary>
-    public class SubmissionOptions
+    public record SubmissionOptions
```

Comment:
Maybe it would be better to remove `Default` and the private constructor, and instead give each init property a default value?